test/acceptance/helpers/snapshot.js

```diff
@@ -0,0 +1,81 @@
+import http from 'http'
+import puppeteer from 'puppeteer'
+
+/**
+ * Similar to our Puppeteer or PercyScript SDKs - uses a Puppeteer page to
+ * inject agent client code and capture the DOM, and sends that DOM with `name`
+ * and `options` to agent's running server.
+ *
+ * Does not use functions defined in sdk-utils to avoid compiling and to allow
+ * errors to bubble up through our tests. Returns the captured DOM string for
+ * assertions within tests.
+ *
+ * @param {object} page - Puppeteer page objectg
+ * @param {string} name - Snapshot name
+ * @param {object} options - Snapshot options
+ * @returns {string} The captured DOM string
+ */
+async function snapshot(page, name, options = {}) {
+  await page.addScriptTag({
+    url: 'http://localhost:5338/percy-agent.js'
+  })
+
+  let dom = await page.evaluate(options => {
+    return new PercyAgent({ handleAgentCommunication: false })
+      .domSnapshot(document, options)
+  }, options)
+
+  let data = JSON.stringify({
+    name,
+    url: page.url(),
+    domSnapshot: dom,
+    clientInfo: 'acceptance-micro-sdk',
```

Comment:
Wonder if we should also pass any node env info?

Reply:
We don't have per-snapshot client info do we? Otherwise our merged builds would show both node versions in the client info section.

Although, these tests never send things off to the real API, so this client info is probably not even necessary. I'll keep it just in case it comes in handy in the future.

Reply:
> Although, these tests never send things off to the real API, so this client info is probably not even necessary. I'll keep it just in case it comes in handy in the future.

Hahaha, very very true. Good point, forgot that. In the real API, we store per-snapshot user agents I believe. Just not surfaced in the UI